src/tests/ftest/util/yaml_utils.py

```diff
@@ -0,0 +1,485 @@
+
+        Args:
+            found (dict): dictionary of matches found for each key
+        """
+        for found_key in found:
+            if found_key not in matches:
+                # Simply add the new value found for this key
+                matches[found_key] = found[found_key]
+
+            else:
+                is_list = isinstance(matches[found_key], list)
+                if not is_list:
+                    matches[found_key] = [matches[found_key]]
+                if isinstance(found[found_key], list):
+                    for found_item in found[found_key]:
+                        if found_item not in matches[found_key]:
+                            matches[found_key].append(found_item)
+                elif found[found_key] not in matches[found_key]:
+                    matches[found_key].append(found[found_key])
+
+                if not is_list and len(matches[found_key]) == 1:
+                    matches[found_key] = matches[found_key][0]
+
+    matches = {}
+    if isinstance(obj, val_type) and isinstance(key, str) and key in keys:
+        # Match found
+        matches[key] = obj
+    elif isinstance(obj, dict):
+        # Recursively look for matches in each dictionary entry
+        for obj_key, obj_val in list(obj.items()):
+            add_matches(find_values(obj_val, keys, obj_key, val_type))
+    elif isinstance(obj, list):
+        # Recursively look for matches in each list entry
+        for item in obj:
+            add_matches(find_values(item, keys, None, val_type))
+
+    return matches
+
+
+def replace_yaml_entries(yaml_file, args):
+    """_summary_.
+
+    Args:
+        yaml_file (str): _description_
+        args (argparse.Namespace): command line arguments for this program
+
+    Raises:
+        CommandFailure: _description_
+
+    Returns:
+        str: modified yaml file
+
+    """
+    log = getLogger()
+    if not os.path.isfile(yaml_file):
+        raise CommandFailure(
+            "Error replacing yaml entries in {}: file does not exist".format(yaml_file))
+
+    yaml_data = get_yaml_data(yaml_file)
+    yaml_keys = list(YAML_KEYS.keys())
+    yaml_find = find_values(yaml_data, yaml_keys, val_type=(str, list, int, dict))
+
+    # Generate a list of values that can be used as replacements
+    user_values = OrderedDict()
+    for key, value in list(YAML_KEYS.items()):
+        args_value = getattr(args, value)
+        if isinstance(args_value, NodeSet):
+            user_values[key] = args_value.copy()
+        elif isinstance(args_value, str):
+            user_values[key] = args_value.split(",")
+        elif args_value:
+            user_values[key] = [args_value]
+        else:
+            user_values[key] = None
+
+    # Assign replacement values for the test yaml entries to be replaced
+    log.info("Detecting replacements for {} in {}".format(yaml_keys, yaml_file))
+    log.info("  Found values: {}".format(yaml_find))
+    log.info("  User values:  {}".format(dict(user_values)))
```

Comment:
(pylint-logging-format-interpolation) Use lazy % formatting in logging functions